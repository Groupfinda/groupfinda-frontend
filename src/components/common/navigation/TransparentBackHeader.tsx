import React from "react";
import {
  Icon,
  TopNavigationAction,
  IconProps,
  TopNavigation,
} from "@ui-kitten/components";
import { StyleSheet } from "react-native";
import { useNavigation } from "@react-navigation/native";

const BackIcon = (props: IconProps) => <Icon {...props} name="arrow-back" />;

type BackActionType = {
  onPress: () => void;
};
const BackAction: React.FC<BackActionType> = ({ onPress }) => {
  return <TopNavigationAction onPress={onPress} icon={BackIcon} />;
};

export default () => {
  const navigation = useNavigation();

  return (
    <TopNavigation
      style={styles.headerStyle}
      accessoryLeft={() => <BackAction onPress={() => navigation.goBack()} />}
    />
  );
};

const styles = StyleSheet.create({
  headerStyle: {
    marginTop: 20,

    position: "absolute",
    backgroundColor: "transparent",
    zIndex: 1,
  },
});
/*
const TopNavigationStyling = () => (
  <TopNavigation
    title={(evaProps) => <Text {...evaProps}>Title</Text>}
    subtitle={(evaProps) => <Text {...evaProps}>Subtitle</Text>}
  />
);
*/
