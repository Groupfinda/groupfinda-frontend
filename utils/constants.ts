export const eventCategories = [
  "Outdoors",
  "Adventure",
  "Technology",
  "Health",
  "Sports",
  "Fitness",
  "Learning",
  "Food & Drink",
  "Culture",
  "Music",
  "Social",
  "Games",
  "Arts",
  "Pets",
  "Hobbies",
  "Volunteer",
];
