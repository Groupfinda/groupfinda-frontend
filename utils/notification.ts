import Constants from 'expo-constants'
import * as Permissions from 'expo-permissions'
import { Platform } from "react-native";
import * as Notifications from 'expo-notifications'

export async function registerForPushNotificationsAsync() {
    let token;
    if (Constants.isDevice) {
        const { status: existingStatus } = await Permissions.getAsync(Permissions.NOTIFICATIONS);
        let finalStatus = existingStatus;
        if (existingStatus !== 'granted') {
            const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
            finalStatus = status;
        }

        if (finalStatus !== 'granted') {
            alert("Notifications allow us to send you updates on group matches and chat messages!")
            return
        }

        token = (await Notifications.getExpoPushTokenAsync()).data
        console.log(token)
    }

    if (Platform.OS === 'android') {
        Notifications.setNotificationChannelAsync('default', {
            name: 'default',
            importance: Notifications.AndroidImportance.MAX,
            vibrationPattern: [0, 250, 250, 250],
            lightColor: "#FF231F7C"
        })
    }

    return token
}