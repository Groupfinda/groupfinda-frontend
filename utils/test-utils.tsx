import React from "react";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { myTheme } from "../custom-theme";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { MockedProvider, MockedResponse } from "@apollo/react-testing";
import { render, RenderAPI, RenderOptions } from "react-native-testing-library";

const AllProviders = (apolloMocks: MockedResponse[]) => ({
  children,
}: {
  children: JSX.Element;
}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...myTheme }}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          {children}
        </MockedProvider>
      </ApplicationProvider>
    </>
  );
};

type CustomRenderType = (
  component: React.ReactElement<any>,
  options?: RenderOptions,
  apolloProps?: MockedResponse[]
) => RenderAPI;

const customRender: CustomRenderType = (
  ui,
  options,
  apolloProps: MockedResponse[] = []
) => render(ui, { wrapper: AllProviders(apolloProps), ...options });

export * from "react-native-testing-library";
export { customRender as render };
