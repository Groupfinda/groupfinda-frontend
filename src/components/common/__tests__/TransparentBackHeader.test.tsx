import React from "react";
import { render, cleanup, fireEvent } from "../../../../utils/test-utils";
import TransparentBackHeader from "../navigation/TransparentBackHeader";

afterEach(cleanup);
const mockfn = jest.fn();
jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({ goBack: mockfn }),
}));

test("renders UI correctly", () => {
  const tree = render(<TransparentBackHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("clicking on pill triggers on press", () => {
  const component = render(<TransparentBackHeader />);
  const button = component.getByTestId("back-button");
  fireEvent.press(button);
  expect(mockfn).toBeCalledTimes(1);
  fireEvent.press(button);
  expect(mockfn).toBeCalledTimes(2);
});
