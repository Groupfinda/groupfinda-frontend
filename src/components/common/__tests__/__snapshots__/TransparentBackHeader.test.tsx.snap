// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "minHeight": 56,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        undefined,
        Object {
          "backgroundColor": "transparent",
          "marginTop": 20,
          "position": "absolute",
          "zIndex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "zIndex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
        testID="back-button"
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="#222B45"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "height": 24,
                  "tintColor": "#222B45",
                  "width": 24,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={4280429381}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGGroup>
                  <RNSVGRect
                    height="24"
                    matrix={
                      Array [
                        6.123233995736766e-17,
                        1,
                        -1,
                        6.123233995736766e-17,
                        24,
                        0,
                      ]
                    }
                    opacity={0}
                    width="24"
                    x={0}
                    y={0}
                  />
                  <RNSVGPath
                    d="M19 11H7.14l3.63-4.36a1 1 0 1 0-1.54-1.28l-5 6a1.19 1.19 0 0 0-.09.15c0 .05 0 .08-.07.13A1 1 0 0 0 4 12a1 1 0 0 0 .07.36c0 .05 0 .08.07.13a1.19 1.19 0 0 0 .09.15l5 6A1 1 0 0 0 10 19a1 1 0 0 0 .64-.23 1 1 0 0 0 .13-1.41L7.14 13H19a1 1 0 0 0 0-2z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View />
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "zIndex": 1,
          },
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          },
        ]
      }
    />
  </View>
</View>
`;
